frontend/main.js

```diff
@@ -321,34 +365,126 @@ async function startApp () {
+                Vue.set(state, k, cheloniaState[k])
+              }
+            })
+            Vue.set(state, 'contracts', cheloniaState.contracts)
+            if (cheloniaState.namespaceLookups) {
+              Vue.set(state, 'namespaceLookups', cheloniaState.namespaceLookups)
+            }
+          // End exclude contracts
+          }
 
-        if (this.$store.state.currentGroupId) {
-          this.initOrResetPeriodicNotifications()
-          this.checkAndEmitOneTimeNotifications()
-        }
-        const databaseKey = `chelonia/persistentActions/${sbp('state/vuex/getters').ourIdentityContractId}`
-        sbp('chelonia.persistentActions/configure', { databaseKey })
-        await sbp('chelonia.persistentActions/load')
+          if (encryptionParams) {
+            sbp('state/vuex/commit', 'login', { identityContractID, encryptionParams })
+          }
+
+          // NOTE: users could notice that they leave the group by someone
+          // else when they log in
+          const currentState = sbp('state/vuex/state')
+          if (!currentState.currentGroupId) {
+            const gId = Object.keys(currentState.contracts)
+              .find(cID => has(currentState[identityContractID].groups, cID))
+
+            if (gId) {
+              sbp('gi.app/group/switch', gId)
+            }
+          }
+
+          // Whenever there's an active session, the encrypted save state should be
+          // removed, as it is only used for recovering the state when logging in
+          sbp('gi.db/settings/deleteEncrypted', identityContractID).catch(e => {
+            console.error('Error deleting encrypted settings after login')
+          })
+
+          this.ephemeral.finishedLogin = 'yes'
 
-        // NOTE: should set IdleVue plugin here because state could be replaced while logging in
-        Vue.use(IdleVue, { store, idleTime: 2 * 60 * 1000 }) // 2 mins of idle config
+          if (this.$store.state.currentGroupId) {
+            this.initOrResetPeriodicNotifications()
+            this.checkAndEmitOneTimeNotifications()
+          }
+          const databaseKey = `chelonia/persistentActions/${sbp('state/vuex/getters').ourIdentityContractId}`
+          sbp('chelonia.persistentActions/configure', { databaseKey })
+          await sbp('chelonia.persistentActions/load')
+
+          // NOTE: should set IdleVue plugin here because state could be replaced while logging in
+          Vue.use(IdleVue, { store, idleTime: 2 * 60 * 1000 }) // 2 mins of idle config
+
+          // TODO: [SW] This should be done by the service worker when logging
+          // in
+          await saveChelonia()
+          sbp('okTurtles.events/emit', LOGIN_COMPLETE, { identityContractID })
+        } catch (e) {
+          sbp('okTurtles.events/emit', LOGIN_ERROR, { identityContractID, error: e })
+        }
       })
       sbp('okTurtles.events/on', LOGOUT, () => {
+        const state = sbp('state/vuex/state')
+        if (!state.loggedIn) return
```

Comment:
Any way to move these additions someplace outside of `main.js`?

Reply:
Done